Protos/generate.sh

```diff
@@ -0,0 +1,189 @@
+# limitations under the License.
+
+set -eu
+
+here="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+root="$here/.."
+protoc=$(which protoc)
+
+# Build the protoc plugins.
+swift build -c release --product protoc-gen-swift
+swift build -c release --product protoc-gen-grpc-swift
+
+# Grab the plugin paths.
+bin_path=$(swift build -c release --show-bin-path)
+protoc_gen_swift="$bin_path/protoc-gen-swift"
+protoc_generate_grpc_swift="$bin_path/protoc-gen-grpc-swift"
+
+# Genreates gRPC by invoking protoc with the gRPC Swift plugin.
+# Parameters:
+# - $1: .proto file
+# - $2: proto path
+# - $3: output path
+# - $4 onwards: options to forward to the plugin
+function generate_grpc {
+  local proto=$1
+  local args=("--plugin=$protoc_generate_grpc_swift" "--proto_path=${2}" "--grpc-swift_out=${3}")
+
+  for option in "${@:4}"; do
+    args+=("--grpc-swift_opt=$option")
+  done
+
+  invoke_protoc "${args[@]}" "$proto"
+}
+
+# Genreates messages by invoking protoc with the Swift plugin.
+# Parameters:
+# - $1: .proto file
+# - $2: proto path
+# - $3: output path
+# - $4 onwards: options to forward to the plugin
+function generate_message {
+  local proto=$1
+  local args=("--plugin=$protoc_gen_swift" "--proto_path=$2" "--swift_out=$3")
+
+  for option in "${@:4}"; do
+    args+=("--swift_opt=$option")
+  done
+
+  invoke_protoc "${args[@]}" "$proto"
+}
+
+function invoke_protoc {
+  # Setting -x when running the script produces a lot of output, instead boil
+  # just echo out the protoc invocations.
+  echo "$protoc" "$@"
+  "$protoc" "$@"
+}
+
+#------------------------------------------------------------------------------
+
+function generate_echo_example {
+  local proto="$here/examples/echo/echo.proto"
+  local output="$root/Sources/Examples/Echo/Model"
+
+  generate_message "$proto" "$(dirname "$proto")" "$output" "Visibility=Public"
+  generate_grpc "$proto" "$(dirname "$proto")" "$output" "Visibility=Public" "TestClient=true"
+}
+
+function generate_routeguide_example {
+  local proto="$here/examples/route_guide/route_guide.proto"
+  local output="$root/Sources/Examples/RouteGuide/Model"
+
+  generate_message "$proto" "$(dirname "$proto")" "$output" "Visibility=Public"
+  generate_grpc "$proto" "$(dirname "$proto")" "$output" "Visibility=Public"
+}
+
+function generate_helloworld_example {
+  local proto="$here/upstream/grpc/examples/helloworld.proto"
+  local output="$root/Sources/Examples/HelloWorld/Model"
+
+  generate_message "$proto" "$(dirname "$proto")" "$output" "Visibility=Public"
+  generate_grpc "$proto" "$(dirname "$proto")" "$output" "Visibility=Public"
+}
+
+function generate_reflection_service {
+  local proto_v1="$here/upstream/grpc/reflection/v1/reflection.proto"
+  local output_v1="$root/Sources/GRPCReflectionService/v1"
+
+  # Messages were accidentally leaked into public API, they shouldn't be but we
+  # can't undo that change until the next major version.
```

Comment:
Are we tracking this?

Reply:
We are now: https://github.com/grpc/grpc-swift/issues/1785